### PR TITLE
[O2B-1040] add RCT periods page

### PR DESCRIPTION
**Pull request by xsalonx**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [ ] issue has "Fix version" assigned
- [x] issue "Status" is set to "In review"
- [x] PR labels are selected

Notable changes for users:
- add `LHC Period ` overview page from RCT

Notable changes for developers:
- add LhcPeriod models

Changes made to the database:
- NA

### CODE REVIEW

**Comment by xsalonx** [comment]
It should be merged after #1224

File: lib/public/Model.js [hunk 1]
```diff
@@ -153,6 +160,9 @@ export default class Model extends Observable {
                 this.logs.fetchAllLogs();
                 this.runs.fetchAllRuns();
                 break;
+            case 'lhc-period-overview':
+                this.lhcPeriods.fetchLhcPeriods();
+                break;
```
Comment (martinboulais) [inline lib/public/Model.js:165]: Please use more recent way of doing stuff: `this.lhcPeriods.loadOverview()` and in it load a sub-model

File: lib/public/views/lhcPeriods/ActiveColumns/lhcPeriodsActiveColumns.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import { h } from '/js/src/index.js';
+
+/**
+ * List of active columns for a generic periods table
+ */
+export const lhcPeriodsActiveColumns = {
+    name: {
+        name: 'Name',
+        visible: true,
+        format: (_, lhcPeriod) => [h('td.text-ellipsis', lhcPeriod.name)],
```
Comment (martinboulais) [inline lib/public/views/lhcPeriods/ActiveColumns/lhcPeriodsActiveColumns.js:23]: Please try to have a look on other tables how it work. You will find out that some tables already have elipsis, and without having to put `td` in `td` 😃

File: lib/public/views/lhcPeriods/overview/LhcPeriodsModel.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+import { Observable, RemoteData } from '/js/src/index.js';
+import { getRemoteDataSlice } from '../../../utilities/fetch/getRemoteDataSlice.js';
+import { PaginationModel } from '../../../components/Pagination/PaginationModel.js';
+
+/**
+ * Model representing handlers for lhc periods page
```
Comment (martinboulais) [inline lib/public/views/lhcPeriods/overview/LhcPeriodsModel.js:19]: Looks like this is the comment that is on some other models, but tbh I think it's really bad. Maybe something like `Model for the LHC period pageS` would be enough.
Aren't you mixing periods and periods statistics again?

File: lib/public/views/lhcPeriods/overview/LhcPeriodsModel.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        return {
+            ...lhcPeriod,
+            ...lhcPeriodData,
+        };
+    }
+
+    /**
+     * Remote data representing all the relevant lhc periods fetched from the API
+     *
+     * @returns {RemoteData[]} the remote data of LHC periods
```
Comment (martinboulais) [inline lib/public/views/lhcPeriods/overview/LhcPeriodsModel.js:93]: It's probably RemoteData of an array, not the other way around

File: lib/public/views/lhcPeriods/overview/LhcPeriodsModel.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+import { Observable, RemoteData } from '/js/src/index.js';
+import { getRemoteDataSlice } from '../../../utilities/fetch/getRemoteDataSlice.js';
+import { PaginationModel } from '../../../components/Pagination/PaginationModel.js';
+
+/**
+ * Model representing handlers for lhc periods page
+ *
+ * @implements {OverviewModel}
+ */
+export class LhcPeriodsModel extends Observable {
```
Comment (martinboulais) [inline lib/public/views/lhcPeriods/overview/LhcPeriodsModel.js:23]: Please use submodels

File: lib/public/views/lhcPeriods/overview/lhcPeriodsPanel.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+import { lhcPeriodsActiveColumns } from '../ActiveColumns/lhcPeriodsActiveColumns.js';
+import { estimateDisplayableRowsCount } from '../../../utilities/estimateDisplayableRowsCount.js';
+import { paginationComponent } from '../../../components/Pagination/paginationComponent.js';
+
+const TABLEROW_HEIGHT = 46;
+// Estimate of the navbar and pagination elements height total; Needs to be updated in case of changes;
+const PAGE_USED_HEIGHT = 215;
+
+/**
+ * The function to load the lhcPeriods overview
```
Comment (martinboulais) [inline lib/public/views/lhcPeriods/overview/lhcPeriodsPanel.js:26]: Same, please avoid copy-paste bad comments...

File: lib/public/views/lhcPeriods/overview/LhcPeriodsModel.js [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+    /**
+     * The constructor of the Overview model object
+     * @param {Model} model Pass the model to access the defined functions
+     * @returns {Object} Constructs the Overview model
+     */
+    constructor(model) {
```
Comment (martinboulais) [inline lib/public/views/lhcPeriods/overview/LhcPeriodsModel.js:29]: Avoid passing model further if you don't need it!

File: lib/public/views/lhcPeriods/overview/lhcPeriodsPanel.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+ * @param {Model} model The overall model object.
+ * @returns {vnode} The overview screen
+ */
+export const LhcPeriodsPage = (model) => {
```
Comment (martinboulais) [inline lib/public/views/lhcPeriods/overview/lhcPeriodsPanel.js:30]: Same comment about the model...

File: test/public/lhcPeriods/overview.test.js [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+
+module.exports = () => {
+    let page;
+    let browser;
+
+    before(async () => {
+        [page, browser] = await defaultBefore(page, browser);
+        await page.setViewport({
+            width: 700,
+            height: 720,
```
Comment (martinboulais) [inline test/public/lhcPeriods/overview.test.js:33]: Do you need this?

**Review by martinboulais — CHANGES_REQUESTED** [review]

File: lib/public/views/lhcPeriods/overview/LhcPeriodsModel.js (hunk 7 above)
Comment (xsalonx) [reply lib/public/views/lhcPeriods/overview/LhcPeriodsModel.js:29]: applied

File: lib/public/Model.js (hunk 1 above)
Comment (xsalonx) [reply lib/public/Model.js:165]: applied

File: lib/public/views/lhcPeriods/ActiveColumns/lhcPeriodsActiveColumns.js (hunk 2 above)
Comment (xsalonx) [reply lib/public/views/lhcPeriods/ActiveColumns/lhcPeriodsActiveColumns.js]: removed

File: lib/public/views/lhcPeriods/overview/LhcPeriodsModel.js (hunk 3 above)
Comment (xsalonx) [reply lib/public/views/lhcPeriods/overview/LhcPeriodsModel.js]: applied

File: lib/public/views/lhcPeriods/overview/LhcPeriodsModel.js (hunk 4 above)
Comment (xsalonx) [reply lib/public/views/lhcPeriods/overview/LhcPeriodsModel.js]: brackets removed

File: test/public/lhcPeriods/overview.test.js (hunk 9 above)
Comment (xsalonx) [reply test/public/lhcPeriods/overview.test.js]: removed

File: lib/public/views/lhcPeriods/overview/lhcPeriodsPanel.js (hunk 6 above)
Comment (xsalonx) [reply lib/public/views/lhcPeriods/overview/lhcPeriodsPanel.js]: rephrased

File: lib/public/views/lhcPeriods/overview/lhcPeriodsPanel.js (hunk 8 above)
Comment (xsalonx) [reply lib/public/views/lhcPeriods/overview/lhcPeriodsPanel.js]: rephrased

File: lib/public/views/lhcPeriods/overview/LhcPeriodsModel.js (hunk 5 above)
Comment (xsalonx) [reply lib/public/views/lhcPeriods/overview/LhcPeriodsModel.js]: done

File: lib/public/views/lhcPeriods/Overview/LhcPeriodsModel.js [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+        this._pagination = new PaginationModel();
+        this._pagination.observe(() => this.fetchLhcPeriods());
+        this._pagination.itemsPerPageSelector$.observe(() => this.notify());
+
+        this._lhcPeriods = RemoteData.NotAsked();
+    }
+
+    /**
+     * Fetch all the relevant periods from the API
+     * @param {Object} filterObject object that defines requested filtering
```
Comment (martinboulais) [inline lib/public/views/lhcPeriods/Overview/LhcPeriodsModel.js:38]: You forgot this bit 😃

File: lib/public/views/lhcPeriods/Overview/lhcPeriodsOverviewPanel.js
Comment (martinboulais) [inline lib/public/views/lhcPeriods/Overview/lhcPeriodsOverviewPanel.js:1]: Why is this file named `...Panel`?

**Review by martinboulais — CHANGES_REQUESTED** [review]

File: lib/public/views/lhcPeriods/Overview/LhcPeriodsModel.js (hunk 10 above)
Comment (xsalonx) [reply lib/public/views/lhcPeriods/Overview/LhcPeriodsModel.js]: refactored

File: lib/public/views/lhcPeriods/Overview/lhcPeriodsOverviewPanel.js
Comment (xsalonx) [reply lib/public/views/lhcPeriods/Overview/lhcPeriodsOverviewPanel.js:1]: renamed

**Review by martinboulais — APPROVED** [review]

**Pull request merged**